dbt/models/iasworld/schema/iasworld.sales.yml

```diff
@@ -202,6 +202,19 @@ sources:
               combination_of_columns:
                 - parid
                 - instruno
+              additional_select_columns:
+                - column: substr(saledt, 1, 4)
+                  label: taxyr
+                  agg_func: max
+                - column: substr(saledt, 1, 4)
+                  label: sale_year
+                  agg_func: array_agg
+                - column: who
+                  label: who
+                  agg_func: array_agg
+                - column: wen
+                  label: wen
+                  agg_func: array_agg
```

Comment:
Which of these two ways of specifying different aggregation functions for `additional_select_columns` do you like better? (The other option being function-specific params like `select_columns_aggregated_with_{func}`, as used in `res_class_matches_pardat` above.) I kind of like the flexibility of this one more, but whichever you prefer, I'll put up a fast-follow PR refactoring the other generic test to use the favored approach.

Reply:
Oh I definitely prefer this one more. In the refactor PR, we may want to include some logic for handling when a mapping is passed with the incorrect columns i.e. column + label, but no agg_func.

Reply:
Great, I just opened an issue to track this here: https://github.com/ccao-data/data-architecture/issues/330